cmd/tracestat/main.go

```diff
@@ -212,38 +227,43 @@ func (ts *tracestat) printCDF() {
 	}
 }
 
-func (ts *tracestat) dumpJSON(f string) {
+func (ts *tracestat) dumpJSON(f string) error {
 	w, err := os.OpenFile(f, os.O_WRONLY|os.O_CREATE, 0644)
 	if err != nil {
-		log.Fatal(err)
+		return err
 	}
 	defer w.Close()
 
 	enc := json.NewEncoder(w)
 
-	type MsgStat struct {
+	// Counts breaks down counts per event type.
+	type Counts struct {
 		Publish, Deliver, Duplicate, Reject, SendRPC, DropRPC int
 	}
 
-	type Sample struct {
-		DelayMillis int
-		Count       int
+	// Bucket represents a bucket in a cumulative distribution function.
+	type Bucket struct {
+		Millis int
+		Count  int
 	}
 
-	type Dump struct {
-		PeerStats       map[string]MsgStat
-		AggregateStats  MsgStat
-		MessageDelays   map[string][]int
-		MessageDelayCDF []Sample
+	// Dump represents the full JSON dump tree.
+	var dump struct {
```

Comment:
Redid this struct to organise the data better. Now we have two levels. Top level is the entity (events, delays), and the second level is the aggregation.